src/Arcus.BackgroundJobs.DataBricks/DatabricksJobMetricsJob.cs

```diff
@@ -0,0 +1,106 @@
+        /// <param name="options">The options to configure the job to query the Databricks report.</param>
+        /// <param name="secretProvider">The instance to provide the token to authenticate with Databricks.</param>
+        /// <param name="logger">The logger instance to to write telemetry to.</param>
+        public DatabricksJobMetricsJob(
+            IOptionsMonitor<DatabricksJobMetricsJobSchedulerOptions> options,
+            ISecretProvider secretProvider,
+            ILogger<DatabricksJobMetricsJob> logger)
+        {
+            Guard.NotNull(options, nameof(options));
+            Guard.NotNull(secretProvider, nameof(secretProvider));
+            Guard.NotNull(logger, nameof(logger));
+
+            DatabricksJobMetricsJobSchedulerOptions value = options.Get(Name);
+            Guard.NotNull(options, nameof(options), "Requires a registered options instance for this background job");
+
+            _options = value;
+            _secretProvider = secretProvider;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// The name of the executing job.
+        /// In order for the <see cref="T:CronScheduler.Extensions.Scheduler.SchedulerOptions" /> options to work correctly make sure that the name is matched
+        /// between the job and the named job options.
+        /// </summary>
+        public string Name { get; } = nameof(DatabricksJobMetricsJob);
+
+        /// <summary>
+        /// This method is called when the <see cref="T:Microsoft.Extensions.Hosting.IHostedService" /> starts. The implementation should return a task that represents
+        /// the lifetime of the long running operation(s) being performed.
+        /// </summary>
+        /// <param name="stoppingToken">Triggered when <see cref="M:Microsoft.Extensions.Hosting.IHostedService.StopAsync(System.Threading.CancellationToken)" /> is called.</param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the long running operations.</returns>
+        public async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            using (DatabricksClient client = await _options.CreateDatabricksClientAsync(_secretProvider))
+            using (var databricksInfoProvider = new DatabricksInfoProvider(client))
+            {
+                DateTimeOffset next = DateTimeOffset.UtcNow;
+                _logger.LogInformation(
+                    "Job monitor for Databricks is starting at {TriggerTime} for time windows {WindowStart} - {WindowEnd}",
+                    DateTimeOffset.UtcNow, _last, next);
+
+                IEnumerable<JobRun> jobRunHistory = await databricksInfoProvider.GetFinishedJobRunsAsync(_last, next);
+                foreach (JobRun jobRun in jobRunHistory)
+                {
+                    ReportJobRunAsMetric(jobRun);
+                }
+
+                _last = next;
+            }
+        }
+
+        private void ReportJobRunAsMetric(JobRun jobRun)
```

Comment:
Suggestion: If we move this to a dedicated class we can use the same logic outside of this job, for example in Azure Functions which we need to support as well.

`databricksInfoProvider.GetFinishedJobRunsAsync(<>,<>)` gets the finished job info
`databricksInfoProvider.ReportFinishedJobOutcomeAsync(JobRun)` reports it as a metric on ILogger
`databricksInfoProvider.MeasureJobOutcomesAsync(<>,<>)` can encapsulate both of these so that `ExecuteAsync` becomes:

```csharp
public async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            using (DatabricksClient client = await _options.CreateDatabricksClientAsync(_secretProvider))
            using (var databricksInfoProvider = new DatabricksInfoProvider(client))
            {
                DateTimeOffset next = DateTimeOffset.UtcNow;
                _logger.LogInformation(
                    "Job monitor for Databricks is starting at {TriggerTime} for time windows {WindowStart} - {WindowEnd}",
                    DateTimeOffset.UtcNow, _last, next);

                await databricksInfoProvider.MeasureJobOutcomesAsync(_last, next);

                _last = next;
            }
        }
```

Names are just indications of course and `DatabricksInfoProvider` might be used by another client on top that provides these methods maybe but you get the idea.

Thoughts?

Reply:
Yes, I'll extrat this to the `DatabricksInfoProvider`.

Reply:
Yes, added the suggested methods to the `DatabricksInfoProvider`.